data/plugins/language_python.lua

```diff
@@ -1,27 +1,66 @@
 -- mod-version:3
 local syntax = require "core.syntax"
 
+
+local python_fstring = {
+
+    patterns = {
+      { pattern = {"{", "}"}, type = "normal", syntax = ".py" },
+      { pattern = "%g",       type = "string"                 },
+      { pattern = "%s",       type = "normal"                 },
+
+    },
+
+    symbols = {}
+}
+
+
+local python_type = {
+
+  patterns = {
+    { pattern = "|",            type = "operator"                           },
+    { pattern = "None",         type = "literal"                            },
+    { pattern = "[%w_]",        type = "keyword2"                           },
+    { pattern = "%s",           type = "normal"                             },
+  },
+
+  symbols = {}
+}
+
+table.insert(python_type.patterns, 1, { pattern = { "%[", "%]" }, syntax = python_type })
+
+
 syntax.add {
   name = "Python",
   files = { "%.py$", "%.pyw$", "%.rpy$", "%.pyi$" },
   headers = "^#!.*[ /]python",
   comment = "#",
   block_comment = { '"""', '"""' },
   patterns = {
-    { pattern = "#.*",                         type = "comment"  },
-    { pattern = { '^%s*"""', '"""' },          type = "comment"  },
-    { pattern = '[uUrR]%f["]',                 type = "keyword"  },
-    { pattern = "class%s+()[%a_][%w_]*",       type = {"keyword", "keyword2"} },
-    { pattern = { '[ruU]?"""', '"""'; '\\' },  type = "string"   },
-    { pattern = { "[ruU]?'''", "'''", '\\' },  type = "string"   },
-    { pattern = { '[ruU]?"', '"', '\\' },      type = "string"   },
-    { pattern = { "[ruU]?'", "'", '\\' },      type = "string"   },
-    { pattern = "-?0[xboXBO][%da-fA-F_]+",type = "number"   },
-    { pattern = "-?%d+[%d%.eE_]*",             type = "number"   },
-    { pattern = "-?%.?%d+",                    type = "number"   },
-    { pattern = "[%+%-=/%*%^%%<>!~|&]",        type = "operator" },
-    { pattern = "[%a_][%w_]*%f[(]",            type = "function" },
-    { pattern = "[%a_][%w_]*",                 type = "symbol"   },
+    { pattern = "#.*",                           type = "comment"                         },
+    { pattern = { '^%s*"""', '"""' },            type = "comment"                         },
+    { pattern = '[uUrR]%f["]',                   type = "keyword"                         },
+    { pattern = "class%s+()[%a_][%w_]*",         type = {"keyword", "keyword2"}           },
+    { pattern = ":%s*[%a_][%w_]*[:%]]",          type = "normal"                          },
+    { pattern = "lambda().*:[%s%a_]",            type = {"keyword, normal"}               },
```

Comment:
I wonder if `.*` can cause problems here. Maybe `.-` would be better?

Reply:
To clarify: if there are multiple `:` in the line, `.*` will take the last one, while `.-` will take the first.